packages/resin-universal-gosquared/src/node.js

```diff
@@ -0,0 +1,37 @@
+var Promise = require('bluebird')
+var GoSquared = require('gosquared')
+
+module.exports = function(gosquaredId, apiKey, debug) {
+	var visitor = null
+	function createVisitor(userId) {
+		var gosquared = new GoSquared({
+			api_key: apiKey,
+			site_token: gosquaredId
+		})
+		if (userId) {
+			visitor = gosquared.createPerson(userId)
+		} else {
+			vistor = gosquared
+		}
+	}
+	function destroyVisitor() {
+		visitor = null
+	}
+	return {
+		login: function(userId) {
+			createVisitor(userId)
+		},
+		logout: function() {
+			destroyVisitor()
+		},
+		track: function(type, data) {
+			// if called before `login` create the object with the random ID
+			createVisitor()
```

Comment:
Currently this will recreate the visitor. It will also drop the userId link which sounds wrong to me